connectors/sources/github.py

```diff
@@ -215,20 +369,297 @@ async def validate_config(self):
     async def close(self):
         await self.github_client.close()
 
-    @retryable(
-        retries=RETRIES,
-        interval=RETRY_INTERVAL,
-        strategy=RetryStrategy.EXPONENTIAL_BACKOFF,
-    )
     async def ping(self):
         try:
             await self.github_client.ping()
             logger.debug("Successfully connected to GitHub.")
-
         except Exception:
             logger.exception("Error while connecting to GitHub.")
             raise
 
+    def prepare_document(self, github_document, schema):
```

Comment:
```suggestion
    def adapt_gh_doc_to_es_doc(self, github_document, schema):
```

nit: I think that method name is a bit more explicit about what it does.